warehouse/query-core/src/main/java/datawave/query/common/grouping/GroupingUtil.java

```diff
@@ -136,21 +136,31 @@ public GroupingInfo getGroupingInfo(Map.Entry<Key,Document> entry, Set<String> g
         return new GroupingInfo(countingMap, fieldVisibilities);
     }
     
-    private Multimap<String,String> getFieldToFieldWithGroupingContextMap(Document d, Set<String> expandedGroupFieldsList,
-                    Map<String,GroupingTypeAttribute<?>> fieldMap, Set<String> groupFieldsSet, Map<String,String> reverseModelMapping) {
+    protected Multimap<String,String> getFieldToFieldWithGroupingContextMap(Document d, Set<String> expandedGroupFieldsList,
+                    Map<String,GroupingTypeAttribute<?>> fieldMap, Set<String> groupFieldsSet, Set<String> contextSet, Map<String,String> reverseModelMapping) {
         
         Multimap<String,String> fieldToFieldWithContextMap = TreeMultimap.create();
         for (Map.Entry<String,Attribute<? extends Comparable<?>>> entry : d.entrySet()) {
             Attribute<?> field = entry.getValue();
             log.trace("field is {}", field);
+
             String fieldName = entry.getKey();
             String shortName = fieldName;
             String shorterName = shortName;
-            if (shortName.indexOf('.') != -1)
-                shortName = shortName.substring(0, shortName.lastIndexOf('.'));
-            if (shorterName.indexOf('.') != -1)
+            boolean containsContext = false;
+
+            if (shortName.indexOf('.') != -1) {
```

Comment:
This method of pulling the grouping context from the fieldname looks a little hokey to me here.  Please see LimitFields.getCommonalityAndGroupingContext(field).  This is pulling off the 2 pieces of the grouping context that are relevant for matching.  LimitFields.removeGrouping(field) will remove the grouping context from the fieldname.  We should either use these methods or move these methods to a common place for use here.